### DISCUSSION TITLE
5 giugno 2023 - esercizio 1

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#38] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/431d646b-01b8-4288-b13e-7228e5c28a05)

**Comment by Warcophyr** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/2e443e69-bd33-4174-96ff-5ad4e85daccb)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/e2843fae-d95c-41ee-8cdb-8c3c0304b468)

**Comment by niccolo-fato** [comment]
![IMAGE 2024-01-14 14:00:09](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/f00e7c71-0f3d-461b-a384-7e6c1699ce59)

**Comment by gioele-m-l** [comment]
> Ogni suggerimento o correzione di eventuali errori è ben accetto/a

1a)
1. Seleziono gli ***Eventi*** di interesse, ovvero quelli compresi tra il 2022 ed il 2023 e il cui costo del biglietto o è compreso tra 80 euro e 150 o è minore di 50 euro, successivamente proietto sull'attributo _Id_ di tali ***Eventi***:

$$EventiValidi = \pi_{Id}(\sigma_{DataInizio \geq 01/01/2022 \land DataFine \leq 31/12/2023 \land ((CostoBiglietto \geq 80 \land CostoBiglietto \leq 150) \lor CostoBiglietto < 50)}(Evento)) $$

2. Il risultato è la proiezione su _Nome_, _Cognome_, _NomeArte_ dei theta join tra ***Artista***, ***Partecipa*** ed ***EventiValidi***:

$$Result = \pi_{Nome, Cognome, NomeArte}(Artista \bowtie_{Artista.Id=Partecipa.IdArtista} Partecipa \bowtie_{Partecipa.IdEvento=EventiValidi.Id} EventiValidi)$$

1b)
1. Creo due tabelle, una per gli ***Artisti*** **non** di "Roma" di cui faccio poi una proiezione su _Id_ e successivamente lo ridenomino in _IdArtista_, l'altra tabella è per gli ***Spettatori*** **non** di "Roma" con passaggi analoghi all'altra tabella:

$$ArtistiNoR = \rho_{IdArtista ← Id}(\pi_{Id}(\sigma_{CittàNascita \neq "Roma"}(Artista)))$$

$$SpettatoriNoR = \rho_{IdSpettatore ← CF}(\pi_{CF}(\sigma_{CittàNascita \neq "Roma"}(Spettatore)))$$

2. Creo delle tabelle di ***Eventi*** non validi (perché partecipati/spettati da non romani) per ognuna delle due tabelle create in precedenza:

$$EvBigliettiNoR = (\pi_{IdEvento}(SpettatoriNoR \bowtie Biglietto)) \bowtie_{IdEvento=Id} Evento$$

$$EvPartecipaNoR = (\pi_{IdEvento}(ArtistiNoR \bowtie Partecipa)) \bowtie_{IdEvento=Id} Evento$$

3. Infine, per trovare il risultato, sottraggo alla tabella ***Evento*** le due tabelle appena create:

$$Result = Evento - EvBigliettiNoR - EvPartecipaNoR$$
